src/legacy/ui/public/state_management/state_hashing/hash_unhash_url.test.ts

```diff
@@ -0,0 +1,299 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { StubBrowserStorage } from 'test_utils/stub_browser_storage';
+import { HashedItemStore } from '../state_storage/hashed_item_store';
+const originalSessionStorage: Storage = (window as any).sessionStorage;
+function mockSessionStorage() {
```

Comment:
Have to do this weird mocking and then lazy require, because have make sure sessionStorage is available before HashedItemSingleton is created. 

As we don't really need in all the test cases features of StubBrowserStorage about `setStubbedSizeLimit`, 
I'd propose to just globally add https://github.com/clarkbw/jest-localstorage-mock to jest instead.

Reply:
Thanks! Indeed a lot better!
It didn't work exactly this way though, had to extract to separate file to help jest figure out the order